File: qc-storage/src/main/java/com/eager/questioncloud/storage/user/UserEntity.java
package com.eager.questioncloud.storage.user;

import com.eager.questioncloud.core.domain.user.model.User;
import com.eager.questioncloud.core.domain.user.vo.UserAccountInformation;
import com.eager.questioncloud.core.domain.user.vo.UserInformation;
import com.eager.questioncloud.core.domain.user.vo.UserStatus;
import com.eager.questioncloud.core.domain.user.vo.UserType;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "user")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long uid;

    @Embedded
    private UserAccountInformation userAccountInformation;

    @Embedded
    private UserInformation userInformation;

    @Column
    @Enumerated(EnumType.STRING)
    private UserType userType;

    @Column
    @Enumerated(EnumType.STRING)
    private UserStatus userStatus;

    @Builder
    public UserEntity(Long uid, UserAccountInformation userAccountInformation, UserInformation userInformation, UserType userType,
        UserStatus userStatus) {
        this.uid = uid;
        this.userAccountInformation = userAccountInformation;
        this.userInformation = userInformation;
        this.userType = userType;
        this.userStatus = userStatus;
    }

    public User toModel() {
        return User.builder()
            .uid(uid)
            .userAccountInformation(userAccountInformation)
            .userInformation(userInformation)
            .userType(userType)
            .userStatus(userStatus)
            .build();
    }

    public static UserEntity from(User user) {
        return UserEntity.builder()
            .uid(user.getUid())
            .userAccountInformation(user.getUserAccountInformation())
            .userInformation(user.getUserInformation())
            .userType(user.getUserType())
            .userStatus(user.getUserStatus())
            .build();
    }
}

File: qc-core/qc-user/qc-user-core/src/main/kotlin/com/eager/questioncloud/user/infrastructure/entity/UserEntity.kt
package com.eager.questioncloud.user.infrastructure.entity

import com.eager.questioncloud.user.domain.User
import com.eager.questioncloud.user.enums.UserStatus
import com.eager.questioncloud.user.enums.UserType
import jakarta.persistence.*

@Entity
@Table(name = "user")
class UserEntity(
    @GeneratedValue(strategy = GenerationType.IDENTITY) @Id var uid: Long = 0,
    @Embedded var userAccountInformationEntity: UserAccountInformationEntity,
    @Embedded var userInformationEntity: UserInformationEntity,
    @Enumerated(EnumType.STRING) @Column var userType: UserType,
    @Enumerated(EnumType.STRING) @Column var userStatus: UserStatus
) {
    fun toModel(): User {
        return User(
            uid,
            userAccountInformationEntity.toModel(),
            userInformationEntity.toModel(),
            userType,
            userStatus
        )
    }

    companion object {
        fun from(user: User): UserEntity {
            return UserEntity(
                user.uid,
                UserAccountInformationEntity.from(user.userAccountInformation),
                UserInformationEntity.from(user.userInformation),
                user.userType,
                user.userStatus
            )
        }
    }
}
